govcd/vdc_test.go

```diff
@@ -487,3 +488,72 @@ func (vcd *TestVCD) TestCreateRawVapp(check *C) {
 	err = task.WaitTaskCompletion()
 	check.Assert(err, IsNil)
 }
+
+func (vcd *TestVCD) TestSetControlAccess(check *C) {
+	// Set VDC sharing to everyone
+	org, err := vcd.client.GetAdminOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	check.Assert(org, NotNil)
+
+	vdc, err := org.GetVDCByName(vcd.config.VCD.Vdc, false)
+	check.Assert(err, IsNil)
+	check.Assert(vdc, NotNil)
+
+	readControlAccessParams, err := vdc.SetControlAccess(true, "ReadOnly", nil, true)
+	check.Assert(err, IsNil)
+	check.Assert(readControlAccessParams, NotNil)
+	check.Assert(readControlAccessParams.IsSharedToEveryone, Equals, true)
+	check.Assert(*readControlAccessParams.EveryoneAccessLevel, Equals, "ReadOnly")
+	check.Assert(readControlAccessParams.AccessSettings, IsNil) // If not shared with users/groups, this will be nil
+
+	// Set VDC sharing to one user
+	orgUserRef := org.AdminOrg.Users.User[0]
+	user, err := org.GetUserByName(orgUserRef.Name, false)
+	check.Assert(err, IsNil)
+	check.Assert(user, NotNil)
+
+	accessSettings := []*types.AccessSetting{
+		{
+			AccessLevel: "ReadOnly",
+			Subject: &types.LocalSubject{
+				HREF: user.User.Href,
+				Name: user.User.Name,
+				Type: user.User.Type,
+			},
+		},
+	}
+
+	readControlAccessParams, err = vdc.SetControlAccess(false, "", accessSettings, true)
+	check.Assert(err, IsNil)
+	check.Assert(readControlAccessParams, NotNil)
+	check.Assert(len(readControlAccessParams.AccessSettings.AccessSetting) > 0, Equals, true)
+	check.Assert(assertVDCAccessSettings(accessSettings, readControlAccessParams.AccessSettings.AccessSetting), IsNil)
+
+	// Check that fail if both isSharedToEveryone and accessSettings is passed
+	readControlAccessParams, err = vdc.SetControlAccess(true, "ReadOnly", accessSettings, true)
```

Comment:
Is it worth it to test also the following use case?
```
readControlAccessParams, err = vdc.SetControlAccess(false, "", nil, true)
```
? (or maybe instead of nil, an empty array?)
Not sure if it would add some value, but just sharing the idea.